### PR TITLE
Add KDF TLS

### PR DESCRIPTION
**Pull request by bfussell**

### CODE REVIEW

File: app/app_main.c [hunk 1]
```diff
@@ -1678,6 +1699,129 @@ static ACVP_RESULT app_hmac_handler(ACVP_TEST_CASE *test_case)
...
+	break;
+    case ACVP_KDF135_TLS_CAP_SHA384:
+        evp_md1 = evp_md2 = EVP_sha384();
+	break;
+    case ACVP_KDF135_TLS_CAP_SHA512:
+        evp_md1 = evp_md2 = EVP_sha512();
+	break;
+    default:
+        printf("\nCrypto module error, Bad SHA type\n");
+        return ACVP_CRYPTO_MODULE_FAIL;
```
Comment (ashman-p) [inline app/app_main.c]: Should this error be something like unsupported algorithm?

File: app/app_main.c [hunk 2]
```diff
@@ -1678,6 +1699,129 @@ static ACVP_RESULT app_hmac_handler(ACVP_TEST_CASE *test_case)
...
+            printf("\nCrypto module error, TLS kdf failure\n");
+            return ACVP_CRYPTO_MODULE_FAIL;
+        }
+        for (i = 0; i < olen1; i++) {
+            master_secret1[i] ^= master_secret2[i];
+        }
+    }
+
+
+    memset(key_block1, 0, 4096);
```
Comment (ashman-p) [inline app/app_main.c]: I guess eventually you may want to add a macro for the magic numbers?

Comment (bfussell) [inline app/app_main.c]: Good catch, I should clear that in the acvp code not the app code where I already have the size.

File: app/app_main.c (hunk 1 above)
Comment (bfussell) [inline app/app_main.c]: Yep, I'll change that

**Pull request merged**